File: 1876-map-of-highest-peak/map-of-highest-peak.java
import java.util.Queue;
class Pair{
    int x ;
    int y ;
    Pair(int x, int y) {
        this.x = x;
        this.y = y;
    }
}
class Solution {
    public int[][] highestPeak(int[][] a) {
        int n = a.length;
        int m = a[0].length;
        Queue<Pair> q = new LinkedList<>();
        int[][] vis = new int[n][m];
        for(int i = 0 ; i < n ; i ++){
            for(int j = 0 ; j < m ; j ++){
                if(a[i][j] == 1){
                    a[i][j] = 1 ;
                    q.offer(new Pair(i, j));
                    vis[i][j] = 1;
                    a[i][j] = 0;
                }
            }
        }
        int[][] dir = {{1, 0}, {-1, 0}, {0, 1}, {0, -1}};
        while(!q.isEmpty()){
            Pair current = q.poll(); 
            int x = current.x;
            int y = current.y;

            for (int[] d : dir) {
                int nx = x + d[0];
                int ny = y + d[1];

                if (nx >= 0 && nx < n && ny >= 0 && ny < m && vis[nx][ny] == 0) {
                    a[nx][ny] = a[x][y] + 1;
                    vis[nx][ny] = 1;
                    q.offer(new Pair(nx, ny));
                }
            }
        }
        return a ;
    }
}

File: 1876-map-of-highest-peak/map-of-highest-peak.py
from queue import Queue
from typing import List

class Solution:
    def highestPeak(self, a: List[List[int]]) -> List[List[int]]:
        n, m = len(a), len(a[0])
        q = Queue()
        
        vis = [[-1 for _ in range(m)] for _ in range(n)] 
        for i in range(n):
            for j in range(m):
                if a[i][j] == 1:
                    q.put((i, j))
                    a[i][j] = 0 
                    vis[i][j] = 0 
        
        dir = [(1, 0), (-1, 0), (0, 1), (0, -1)]
        
        while not q.empty():
            x, y = q.get()
            for dx, dy in dir:
                nx, ny = x + dx, y + dy
                if 0 <= nx < n and 0 <= ny < m and vis[nx][ny] == -1:
                    a[nx][ny] = a[x][y] + 1
                    vis[nx][ny] = a[nx][ny]
                    q.put((nx, ny))
        
        return a